wolfcrypt/src/ecc.c

```diff
@@ -4070,9 +4076,16 @@ static int wc_ecc_shared_secret_gen_sync(ecc_key* private_key, ecc_point* point,
         wc_ecc_del_point_ex(result, private_key->heap);
     }
 #endif
+
+  errout:
+
 #ifdef HAVE_ECC_CDH
-    if (k == &k_lcl)
+    if (k == k_lcl)
         mp_clear(k);
+#ifdef WOLFSSL_SMALL_STACK
+    if (k_lcl != NULL)
+        XFREE(k_lcl, private_key->heap, DYNAMIC_TYPE_ECC_BUFFER);
```

Comment:
since this is private key data should there be some sort secure memzero() or something?

Reply:
this is a bit subtle (had to review the code a couple times to grok it) but it turns out to be OK as-is:

if `k_lcl` is non-null, then it will always be true that `k == k_lcl`, hence `mp_clear(k)` will always be done before the `XFREE()`.